enqueue.go

```diff
@@ -14,6 +14,20 @@ const (
 	NanoSecondPrecision = 1000000000.0
 )
 
+type Producer struct {
```

Comment:
What do you think about renaming this file to `producer.go`?

Reply:
I meant to, and didn't.